### PR TITLE
Shot Data Inferred Stats

### PR DESCRIPTION
**Pull request by tsnl**
Solution for #3

---

- Implemented `notebooks/ShotDataInferredStats.jl.ipynb` computing inferred statistics.
  This emits the file `data/shot_data_aug.csv`.
- Modified `scripts/ExtractShotData.jl` to emit either JSON or CSV files. 
  Normalized file-paths to work on my machine too.
- Added `scripts/Setup.jl` to install dependencies.

---

Changes to data pipeline
- All scripts depend on the [StatsBomb Open-Data Dataset](https://github.com/statsbomb/open-data) being located at `/data/open-data-master`. This replaces hard-coded paths that were unique to @nimberledge's file-system.
- `ExtractShotData.jl` now produces either CSV or JSON, accepts different arguments. Example of new usage:
  
  ```
  $ julia scripts/ExtractShotData.jl data/shot_data.json     # outputs the JSON file 'data/shot_data.json'
  $ julia scripts/ExtractShotData.jl data/shot_data.csv      # outputs the CSV file 'data/shot_data.csv
  ```

### CODE REVIEW

**Comment by tsnl** [comment]
@nimberledge how do you recommend I test this?

File: scripts/ExtractShotData.jl [hunk 1]
```diff
@@ -1,41 +1,88 @@
...
 "Trim down the shot data to only the parameters we're interested in for now."
 function trim_shot_data(events)
     retval = []
     for event in events
-        tdict = Dict()
-        tdict["location"] = event["location"]
-        tdict["body_part"] = event["shot"]["body_part"]["name"]
-        tdict["outcome"] = event["shot"]["outcome"]["name"]
-        tdict["type"] = event["shot"]["type"]["name"]
+        location = event["location"]
```
Comment (nimberledge) [inline scripts/ExtractShotData.jl:17]: I actually don't want to filter at this stage, but this works for now. If we're gonna do this, would rather we compared lowercase(type) to lowercase("Open Play")

File: data-visualization/FBPlots.jl [hunk 2]
```diff
@@ -1,3 +1,5 @@
+include!("../data-utils/Utils.jl")
```
Comment (nimberledge) [inline data-visualization/FBPlots.jl]: Why? As far as I know, this is unnecessary. Unless you had some future design ideas in mind

File: data-utils/Utils.jl [hunk 3]
```diff
@@ -1,6 +1,18 @@
 import JSON
 import PrettyPrint
 
+statsbomb_events_data = "/data/open-data-master/data/events/"
```
Comment (nimberledge) [inline data-utils/Utils.jl]: Noted, will put in the field spec info in a future commit

File: notebooks/ShotDataInferredStats.jl.ipynb [hunk 4]
```diff
@@ -0,0 +1,410 @@
+{
```
Comment (nimberledge) [inline notebooks/ShotDataInferredStats.jl.ipynb:1]: Just checked out the branch and tried to run NB, 

Couple notes on the whole (it's a pain to locate code locs in the .ipynb format)

1. There's like a couple typos (input_file_path vs file_path in the df initialization cell)
2. Run NB start to finish to get it in a state to commit
3. Not that it's at all a big deal rn but float64 is totally overkill, we can use float32 with no real downside
4. I don't think you're actually using any BLAS functionality here

**Review by nimberledge — CHANGES_REQUESTED** [review]
Thank you so much for doing this, really is a big help. Especially with the Setup script and extra repo organization. This started as a personal project but it's lovely to have some more pro design principles being employed here. <3

File: data-visualization/FBPlots.jl (hunk 2 above)
Comment (tsnl) [reply data-visualization/FBPlots.jl]: Moved geometry constants into there

**Review by nimberledge — APPROVED** [review]
Sorree I had to change a bunch more than the ticket defined :P

**Pull request merged**